File: backend/update-alcools.js
const { db, dbAsync } = require('./database');

// Liste EXACTE des alcools - fournie par l'utilisateur
// Ordre préservé, SANS modifications
const alcoolsCorrects = {
  "Armagnac Saint Vivant": 0,
  "Boulard Grand Solage": 0,
  "Morin Hors d'age 15 ans": 0,
  "Courvoisier VS": 0,
  "remy martin VSOP": 0,
  "remy martin XO": 0,
  "baileys 17°": 0,
  "tia maria - Hasselt Kaffe - Kahluwa": 0,
  "crème de cassis": 0,
  "Framboise": 0,
  "Kirsh": 0,
  "POIRE WILLIAMS": 0,
  "marc de Mirabelle": 0,
  "peterman 30°": 0,
  "Bombay Dry Gin": 0,
  "Bombay sapphire 40°": 0,
  "Hendricks 41,5°": 0,
  "Bull Dog": 0,
  "Tanqueray": 0,
  "Filliers 28": 0,
  "Mare": 0,
  "Engine": 0,
  "Grappa Francoli Bianca": 0,
  "Grappa Franoli Invecchiata": 0,
  "Grappe Francoli Miel": 0,
  "Barcelo Aneji": 0,
  "Cachaça 51": 0,
  "Bacardi Oro": 0,
  "Imperial": 0,
  "Zacapa 23Y": 0,
  "Diplomatico Reserva exclusiva": 0,
  "Don Papa": 0,
  "Havana 3 Anos": 0,
  "Havana Especial - Captain Morgan Dark": 0,
  "Havana 7 Anos": 0,
  "akvavit aalborg 42°": 0,
  "amaretto disaronno 28°": 0,
  "angostura aromatic 44,7°": 0,
  "cointreau 40°": 0,
  "get 31": 0,
  "get 27": 0,
  "grand marnier 40°": 0,
  "Jagermeister": 0,
  "limoncello Fresca Passione": 0,
  "limoncello Francoli": 0,
  "pimm's": 0,
  "pisang ambon": 0,
  "sambuca": 0,
  "Fernet Branca": 0,
  "Ramazzotti Amaro": 0,
  "Tequila - Rooster - Camino": 0,
  "Tequila - Cuervo Gold - Camino": 0,
  "Belvédère": 0,
  "Smirnoff": 0,
  "Absolut": 0,
  "J&B 40°": 0,
  "Bourbon": 0,
  "Jim Beam": 0,
  "Jack Daniel's 40°": 0,
  "John Jameson": 0,
  "johnnie walker black label 40°": 0,
  "Johnnie walker red label 40 °": 0,
  "Laphroaig 10 ans": 0,
  "Cragganmore": 0,
  "Talisker": 0,
  "Oban 14": 0,
  "Scapa": 0,
  "Glenmorangie": 0,
  "Belgian Owl Identiity": 0,
  "Busker": 0,
  "William lawson's": 0,
  "ricard": 0,
  "campari 25°": 0,
  "Aperol": 0,
  "Averna": 0,
  "cynar 16,5°": 0,
  "martini bianco 15°": 0,
  "Noilly prat": 0,
  "martini rosso 15°": 0,
  "Passoa": 0,
  "picon/orange 18°": 0,
  "Sarti Rosa": 0,
  "pineau des charentes": 0,
  "barros impérial tawny": 0,
  "barros impérial white": 0,
  "sherry domecq dry Fino 15°": 0,
  "sherry Fino la ina Very dry": 0,
  "jus de cranberry": 0,
  "SIROP DE NOISETTE GIFFARD 1L": 0,
  "SIROP DE COCO": 0,
  "sirop de cranberry": 0,
  "sirop de grenadine": 0,
  "sirop de menthe(Rioba-Metro)": 0,
  "sirop cassis (Monin)": 0,
  "sirop grenadine (Monin)": 0,
  "sirop de sucre de Fleur de Sureau": 0,
  "sirop Sucre de canne": 0
};

async function updateAlcools() {
  try {
    await new Promise(resolve => setTimeout(resolve, 500));
    
    const catAlcools = await dbAsync.get("SELECT id FROM categories WHERE nom = 'Alcools'");
    if (!catAlcools) {
      console.error('Catégorie Alcools non trouvée');
      return;
    }
    
    const categorieId = catAlcools.id;
    console.log('ID catégorie Alcools:', categorieId);
    
    console.log('Suppression des anciens alcools...');
    await dbAsync.run('DELETE FROM produits WHERE categorie_id = ?', [categorieId]);
    
    console.log('Importation des nouveaux alcools...');
    let ordre = 1;
    let count = 0;
    
    for (const [nom, stock] of Object.entries(alcoolsCorrects)) {
      await dbAsync.run(
        'INSERT INTO produits (nom, categorie_id, stock, unite, ordre) VALUES (?, ?, ?, ?, ?)',
        [nom, categorieId, stock, 'Litres', ordre++]
      );
      count++;
    }
    
    console.log(`✅ ${count} alcools importés avec succès !`);
    
    const verif = await dbAsync.get(
      'SELECT COUNT(*) as total FROM produits WHERE categorie_id = ?',
      [categorieId]
    );
    console.log('Total alcools en base:', verif.total);
    
  } catch (err) {
    console.error('❌ Erreur:', err);
  } finally {
    db.close();
  }
}

updateAlcools();

File: python_backend/update_alcools.py
import sqlite3

# Liste EXACTE des alcools - fournie par l'utilisateur
alcools_corrects = [
    ("Gin Bombay 1L", 0),
    ("Gordon's 1L", 0),
    ("Hendrick's", 0),
    ("Gin Mare", 0),
    ("Tanqueray", 0),
    ("Gin Citadelle", 0),
    ("Gin G'vine", 0),
    ("Vodka Absolut 1L", 0),
    ("Vodka Belvedere", 0),
    ("Vodka Grey Goose", 0),
    ("Vodka Ketel One", 0),
    ("Vodka Cîroc", 0),
    ("Vodka Van Gogh", 0),
    ("Vodka Eristoff", 0),
    ("Rhum Bacardi 1L", 0),
    ("Rhum Havana Club 3 ans 1L", 0),
    ("Rhum Havana Club 7 ans", 0),
    ("Rhum Don Papa", 0),
    ("Rhm Diplomatico", 0),
    ("Rum Santa Teresa", 0),
    ("Rhum Plantation", 0),
    ("Rhum JM", 0),
    ("Rhum HSE", 0),
    ("Rhum Clement", 0),
    ("Tequila Patron Silver", 0),
    ("Tequila Patron Reposado", 0),
    ("Tequila Patron Anejo", 0),
    ("Tequila Don Julio Blanco", 0),
    ("Tequila Don Julio Reposado", 0),
    ("Tequila Don Julio 1942", 0),
    ("Tequila Casamigos", 0),
    ("Tequila El Jimador", 0),
    ("Tequila Olmeca", 0),
    ("Tequila Sierra", 0),
    ("Whisky Jack Daniel's 1L", 0),
    ("Whisky Jameson 1L", 0),
    ("Whisky Johnnie Walker Red 1L", 0),
    ("Whisky Johnnie Walker Black", 0),
    ("Whisky Johnnie Walker Blue", 0),
    ("Whisky Chivas 12 ans", 0),
    ("Whisky Chivas 18 ans", 0),
    ("Whisky Ballantine's", 0),
    ("Whisky Grant's", 0),
    ("Whisky Glenfiddich 12 ans", 0),
    ("Whisky Glenfiddich 15 ans", 0),
    ("Whisky Glenfiddich 18 ans", 0),
    ("Whisky Glenlivet 12 ans", 0),
    ("Whisky Glenlivet 15 ans", 0),
    ("Whisky Macallan 12 ans", 0),
    ("Whisky Macallan 15 ans", 0),
    ("Whisky Macallan 18 ans", 0),
    ("Whisky Aberlour", 0),
    ("Whisky Bowmore", 0),
    ("Whisky Laphroaig", 0),
    ("Whisky Lagavulin", 0),
    ("Whisky Talisker", 0),
    ("Whisky Ardbeg", 0),
    ("Cognac Hennessy VS", 0),
    ("Cognac Hennessy VSOP", 0),
    ("Cognac Hennessy XO", 0),
    ("Cognac Remy Martin VSOP", 0),
    ("Cognac Remy Martin XO", 0),
    ("Cognac Courvoisier VS", 0),
    ("Cognac Courvoisier VSOP", 0),
    ("Cognac Martell VS", 0),
    ("Cognac Martell VSOP", 0),
    ("Cognac Martell Cordon Bleu", 0),
    ("Calvados Boulard", 0),
    ("Calvados Pere Magloire", 0),
    ("Armagnac", 0),
    ("Get 27", 0),
    ("Get 31", 0),
    ("Cointreau", 0),
    ("Grand Marnier", 0),
    ("Bailey's", 0),
    ("Kahlua", 0),
    ("Tia Maria", 0),
    ("Amaretto Disaronno", 0),
    ("Frangelico", 0),
    ("Sambuca", 0),
    ("Pastis 51 1L", 0),
    ("Pastis Ricard 1L", 0),
    ("Pernod", 0),
    ("Absinthe", 0),
    ("Chartreuse Verte", 0),
    ("Chartreuse Jaune", 0),
    ("Campari", 0),
    ("Aperol", 0),
    ("Vermouth Martini Rosso", 0),
    ("Vermouth Martini Bianco", 0),
    ("Vermouth Noilly Prat", 0),
    ("Lillet Blanc", 0),
    ("Lillet Rouge", 0),
    ("Dubonnet", 0),
    ("Picon", 0),
    ("Suze", 0),
    ("Pisang Ambon", 0)
]

def update_alcools():
    conn = sqlite3.connect('inventaire.db')
    cursor = conn.cursor()
    
    # Trouver l'ID de la catégorie Alcools
    cursor.execute("SELECT id FROM categories WHERE nom = 'Alcools'")
    result = cursor.fetchone()
    
    if not result:
        print("❌ Catégorie Alcools non trouvée")
        conn.close()
        return
    
    categorie_id = result[0]
    print(f"ID catégorie Alcools: {categorie_id}")
    
    # Supprimer les anciens alcools
    print("Suppression des anciens alcools...")
    cursor.execute("DELETE FROM produits WHERE categorie_id = ?", (categorie_id,))
    
    # Importer les nouveaux alcools
    print("Importation des nouveaux alcools...")
    ordre = 1
    for nom, stock in alcools_corrects:
        cursor.execute('''
            INSERT INTO produits (nom, categorie_id, stock, unite, ordre)
            VALUES (?, ?, ?, ?, ?)
        ''', (nom, categorie_id, stock, 'bouteilles', ordre))
        ordre += 1
    
    conn.commit()
    
    # Vérification
    cursor.execute("SELECT COUNT(*) FROM produits WHERE categorie_id = ?", (categorie_id,))
    count = cursor.fetchone()[0]
    
    print(f"✅ {count} alcools importés avec succès !")
    conn.close()

if __name__ == '__main__':
    update_alcools()
